subsys/sd/sd.c

```diff
@@ -127,7 +127,9 @@ static int sd_init_io(struct sd_card *card)
 	bus_io->bus_width = SDHC_BUS_WIDTH1BIT;
 	/* Cards start with legacy timing and 3.3V signalling at power on */
 	bus_io->timing = SDHC_TIMING_LEGACY;
-	bus_io->signal_voltage = SD_VOL_3_3_V;
+	bus_io->signal_voltage = card->host_props.host_caps.vol_330_support ?
```

Comment:
Please update the comment here, and correctly record the signal voltage below: https://github.com/zephyrproject-rtos/zephyr/blob/18636af6d6658cf66e5f055901ffe2515116d329/subsys/sd/sd.c#L148